src/benchmarks/micro/libraries/System.Collections/Indexer/IndexerSet.cs

```diff
@@ -64,6 +69,30 @@ public List<T> List()
             return list;
         }
 
+        [GlobalSetup(Target = nameof(ImmutableArray))]
+        public void SetupImmutableArray() => _immutableArray = Immutable.ImmutableArray.CreateRange(ValuesGenerator.ArrayOfUniqueValues<T>(Size));
+
+        [Benchmark]
+        public ImmutableArray<T> ImmutableArray()
+        {
+            var immutableArray = _immutableArray;
+            for (int i = 0; i < immutableArray.Length; i++)
+                immutableArray = immutableArray.SetItem(i, default);
+            return immutableArray;
+        }
+
+        [GlobalSetup(Target = nameof(ImmutableList))]
+        public void SetupImmutableList() => _immutableList = Immutable.ImmutableList.CreateRange(ValuesGenerator.ArrayOfUniqueValues<T>(Size));
+
+        [Benchmark]
+        public ImmutableList<T> ImmutableList()
+        {
+            var immutableList = _immutableList;
+            for (int i = 0; i < immutableList.Count; i++)
+                immutableList = immutableList.SetItem(i, default);
```

Comment:
SetItem is not technically an indexer set, but in other benchmarks (e. g. the Ctor benchmarks) there was precedent for including the immutable equivalent of a mutable method. So, I felt like it made sense to include this here rather than creating a separate SetItem benchmark. Let me know if you disagree.